drivers/flash/flash_mcux_flexspi.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT	nxp_imx_flexspi
+
+#include <drivers/flash.h>
+#include <logging/log.h>
+#include <sys/util.h>
+#include "flash_mcux_flexspi.h"
+
+LOG_MODULE_REGISTER(flash_flexspi, CONFIG_FLASH_LOG_LEVEL);
+
+struct flash_flexspi_config {
+	FLEXSPI_Type *base;
+	uint8_t *ahb_base;
+	bool ahb_bufferable;
+	bool ahb_cacheable;
+	bool ahb_prefetch;
+	bool ahb_read_addr_opt;
+	bool combination_mode;
+	bool xip;
+	flexspi_read_sample_clock_t rx_sample_clock;
+};
+
+struct flash_flexspi_data {
+	size_t size[kFLEXSPI_PortCount];
+};
+
+int flash_flexspi_update_lut(const struct device *dev, uint32_t index,
+		const uint32_t *cmd, uint32_t count)
+{
+	const struct flash_flexspi_config *config = dev->config;
+
+	FLEXSPI_UpdateLUT(config->base, index, cmd, count);
+
+	return 0;
+}
+
+int flash_flexspi_set_flash_config(const struct device *dev,
+		const flexspi_device_config_t *device_config,
+		flexspi_port_t port)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	struct flash_flexspi_data *data = dev->data;
+
+	if (port >= kFLEXSPI_PortCount) {
+		LOG_ERR("Invalid port number");
+		return -EINVAL;
+	}
+
+	data->size[port] = device_config->flashSize * KB(1);
+
+	FLEXSPI_SetFlashConfig(config->base,
+			       (flexspi_device_config_t *) device_config,
+			       port);
+
+	return 0;
+}
+
+int flash_flexspi_reset(const struct device *dev)
+{
+	const struct flash_flexspi_config *config = dev->config;
+
+	FLEXSPI_SoftwareReset(config->base);
+
+	return 0;
+}
+
+int flash_flexspi_transfer(const struct device *dev,
+		flexspi_transfer_t *transfer)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	status_t status = FLEXSPI_TransferBlocking(config->base, transfer);
+
+	if (status != kStatus_Success) {
+		LOG_ERR("Transfer error: %d", status);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+void *flash_flexspi_get_ahb_address(const struct device *dev,
+		flexspi_port_t port, off_t offset)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	struct flash_flexspi_data *data = dev->data;
+	int i;
+
+	if (port >= kFLEXSPI_PortCount) {
+		LOG_ERR("Invalid port number");
+		return NULL;
+	}
+
+	for (i = 0; i < port; i++) {
+		offset += data->size[port];
+	}
+
+	return config->ahb_base + offset;
+}
+
+static int flash_flexspi_init(const struct device *dev)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	flexspi_config_t flexspi_config;
+
+	if (config->xip) {
+		LOG_ERR("XIP not supported");
+		return -EINVAL;
+	}
+
+	FLEXSPI_GetDefaultConfig(&flexspi_config);
+
+	flexspi_config.ahbConfig.enableAHBBufferable = config->ahb_bufferable;
```

Comment:
Since we can't "fix" this sort of thing in edtlib, an alternative is to invert the devicetree property meaning. Something like `no-ahb-bufferable` in devicetree, and `.ahb_bufferable = !DT_INST_PROP(..., no_ahb_bufferable)` when initializing `config`.

Feel free to disregard if you're happier with setting this in .dtsi.

Reply:
I think it would be confusing to tie a property to hardware register field but invert the logic, so keeping this in the dtsi. Thanks for your help